docs/designs/data-models.md

```diff
@@ -0,0 +1,116 @@
+# Data Models Dictionary
+
+This document presents the data model used in this software. It describes different domain entities, fields within
+them, and relationships among the entities. Reading this document should provide a good overview of how the 
+product should work under-the-hood.
+
+Before you proceed, please read [terminologies.md](../terminologies.md) document to familiarize yourself with the 
+terms.
+
+## Entities
+This section describes the entites and fields within each of them.
+
+### User (aka Account)
+The person who uses the product and signed up for an account. The User entity also contains credentials for the account
+they created within the product.
+
+* **First Name**
+* **Last Name**
+* **Email**
+* **Password**
+* **Organization**: Which organization is this person affiliated with?
+* **Workspaces**: Workspaces owned by this user.  Currently we only support one workspace for a user
+
+### Organization
+A non-profit organization, company, university or any institution the user is affiliated with.
+There is a special organization called *__individual__* used to represent users who are not affiliated with any 
+organization.
+
+* **Name**
+* **Website**
+* **Disease**
+
+### Disease
+Information about the disease represented by the organization or the user
+
+* **Name**
+* **OMIM ID**: OMIM is an online database that assigns an ID for diseases. Ex: https://omim.org/entry/250220
+
+### Workspace
+Workspace is a logical grouping of Programs within the product.
+
+* **Name**: (unused)
+* **Programs**: List of programs within this workspace
+
+### Program
+A program represents a track of work to create a treatment
+
+* **Name**
+* **Description**
+* **Projects**: Collection of projects that are a part of the program
+
+### Project
+A Project represents an activity necessary to advance the Program forward. A Project is a collection of Tasks
+
+* **Name**
+* **Description**
+* **Status**: Is this project active, or deleted or archived?
+* **Tasks**: Collection of Tasks that are a part of the project
+* **Education**: Education materials to help users understand the purpose of the Project and guide them to make a decision 
+
+### Task
+A Task is the smallest unit of work from a planning point-of-view. 
+
+* **Name**
+* **Description**
+* **Status**: What is the status of this task? In-Progress, Completed etc
+* **Asignee**: Who is responsible for completing the task?
+* **Budget**: Expected Cost of completing this task
```

Comment:
How about removing this and adding a 'metadata' field? All tasks do not necessarily need to have a budget associated with them. Also, some users may want to view additional fields in the UI for each and every task.